src/core/api.proto

```diff
@@ -94,6 +94,13 @@ message InferRequestHeader
     //@@       for tensors with a non-fixed-size datatype (like STRING).
     //@@
     uint64 batch_byte_size = 3;
+
+    //@@    .. cpp:var:: string shm_name
+    //@@
+    //@@       The name given by the user during registration of a shared memory
+    //@@       region that holds the input data.
+    //@@
+    string shm_name = 4;
```

Comment:
We still need the offset and size. User can register a single shared memory region and use (parts of) it for different inputs and outputs, even across multiple requests. So I thing you still need the InferSharedMemory message but the with "key" being the registered shared memory region name.

Reply:
Understood